### PR TITLE
Finish up taxi demo

### PR DESCRIPTION
**Pull request by rikturr**
- Ran all models for tip percentage prediction. 
- Tweak dashboard code to read/write all files from S3 instead of repo. 
- Create pared down ML deployment script
- Summary notebook with description and timings

There is a lot here, and I verified that all the pieces work together. There are still outstanding items around documentation that can be outside the scope of this review. I would like if you all can take a look and double-check the `summary.ipynb` to make sure all descriptive text is correct, as well as analysis of the different ML model runtimes.

https://taxi-dashboard-deploy.demo.saturnenterprise.io/ for finished product!

### CODE REVIEW

File: taxi_demo/summary.ipynb [hunk 1]
```diff
@@ -0,0 +1,1050 @@
...
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import dask.dataframe as dd\n",
+    "import holoviews as hv\n",
+    "import hvplot.pandas"
+   ]
+  },
+  {
+   "cell_type": "code",
```
Comment (jsignell) [inline taxi_demo/summary.ipynb:64]: This cell is not needed.

File: taxi_demo/summary.ipynb [hunk 2]
```diff
@@ -0,0 +1,1050 @@
...
+    "| Dask | CPU cluster |\n",
+    "| Spark | CPU cluster* |\n",
+    "| RAPIDS | GPU cluster |\n",
+    "\n",
+    "Models:\n",
+    "- Elastic net regression + hyperparameter tuning\n",
+    "- XGBoost regression\n",
+    "- Random forest classification\n",
+    "\n",
+    "Not all models were run with all libraries for various reasons, this matrix records which were run:\n",
```
Comment (jsignell) [inline taxi_demo/summary.ipynb:110]:
```suggestion
    "Not all models have been run yet with all libraries, this matrix records which have already been run:\n",
```

File: taxi_demo/summary.ipynb [hunk 3]
```diff
@@ -0,0 +1,1050 @@
+{
```
Comment (jsignell) [inline taxi_demo/summary.ipynb:1]: Either restart the kernel and re-run the whole notebook before checking it in, or clear all the output. It looks sketchy to have out of order cell input numbers.

File: taxi_demo/summary.ipynb (hunk 1 above)
Comment (rikturr) [reply taxi_demo/summary.ipynb:64]: it doesnt render on github, but this left-justifies the table, which i like in jupyterlab

Comment (rikturr) [reply taxi_demo/summary.ipynb:64]: im referring to markdown here, not cell output

**Pull request merged**